### PR TITLE
Add tests for -XX:[+|-]ClassRelationshipVerifier

### PR DESCRIPTION
**Pull request by sharon-wang**
Test the following cases when `-XX:+ClassRelationshipVerifier` is used:
- only source class loaded
- only target class loaded
- neither target nor source classes loaded
- source is subclass of target
- source implements target
- source is not subclass of target
- source does not implement target

Related: https://github.com/eclipse/openj9/pull/7089

Extensions changes:
- JDK: https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/139
- JDK 8: https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/328
- JDK 11: https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/211
- ~JDK 12: https://github.com/ibmruntimes/openj9-openjdk-jdk12/pull/72~ not needed
- JDK 13: https://github.com/ibmruntimes/openj9-openjdk-jdk13/pull/14

### CODE REVIEW

**Comment by sharon-wang** [comment]
I added this test alongside the existing `bcverify bcuwhite` test - not sure if this is the best place for the new test - happy to move it if it belongs something else!

File: runtime/tests/bcverify/relationships/bcvrelationships.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+Java_org_openj9_test_classRelationshipVerifier_TestClassRelationshipVerifier_isRelationshipRecorded(JNIEnv *env, jclass clazz, jstring childNameString, jstring parentNameString, jobject classLoaderObject)
+{
+	jboolean isRelationshipRecorded = JNI_FALSE;
+	J9VMThread *currentThread  = (J9VMThread *) env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	J9ClassLoader *classLoader = NULL;
+	U_8 *childName = NULL;
+	U_8 *parentName = NULL;
+	UDATA childNameLength = 0;
```
Comment (sharon-wang) [inline runtime/tests/bcverify/relationships/bcvrelationships.c]: can move these to a more inner scope

File: runtime/include/j9lib.h.in [hunk 2]
```diff
@@ -65,6 +65,7 @@ extern"C"{
 #define J9_TOOLS_OS_MEMORY_DLL_NAME "osmemory29"
 #define J9_TOOLS_SOFTMX_TEST_DLL_NAME "softmxtest"
 #define J9_TOOLS_BALLOON_DLL_NAME "balloon29"
+#define J9_VERIFIER_TEST_RELATIONSHIPS_DLL_NAME "bcvrelationships"
```
Comment (keithc-ca) [inline runtime/include/j9lib.h.in]: This doesn't appear to be used. Please revert this change.

File: test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/CustomClassLoader.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package org.openj9.test.classRelationshipVerifier;
+
+class CustomClassLoader extends ClassLoader {
+	public Class<?> getClass(String name, byte[] bytes){
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/CustomClassLoader.java]: formatting: space before `{`

File: test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/TestClassRelationshipVerifier.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+		byte[] bytesClassB = ClassGenerator.classB_dump();
+
+		Class<?> clazzA = classLoader.getClass("A", bytesClassA);
+		Class<?> clazzB = classLoader.getClass("B", bytesClassB);
+
+		try {
+			Object instanceB = clazzB.getDeclaredConstructor().newInstance();
+			Object instanceA = clazzA.getDeclaredConstructor().newInstance();
+
+			if (!isRelationshipRecorded("B", "C", classLoader)){
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/TestClassRelationshipVerifier.java]: formatting: space before `{`; here and on lines 97 & 122

**Comment by keithc-ca** [comment]
The PR for Java 12 is not needed - I closed it.

**Comment by sharon-wang** [comment]
@keithc-ca Thanks for reviewing! Added the requested changes and travis build passed.

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux all

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk8,jdk11,jdk13

**Comment by keithc-ca** [comment]
Jenkins test sanity win32 jdk8

**Comment by keithc-ca** [comment]
There are windows 32-bit link errors:
```
00:15:47.876     Creating library ../../../lib/bcvrelationships.lib and object ../../../lib/bcvrelationships.exp
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_kill referenced in function _hashTableFree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _avl_search referenced in function _hashTableFindNodeInTree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_removeElement referenced in function _hashTableAddNodeInTree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _avl_insert referenced in function _hashTableAddNodeInTree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_newElement referenced in function _hashTableAddNodeInTree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_ensureCapacity referenced in function _listToTree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _avl_delete referenced in function _hashTableRemoveNodeInTree
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_clear referenced in function _rebuildFromPools
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_nextDo referenced in function _rebuildFromPools
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_startDo referenced in function _rebuildFromPools
00:15:47.876  j9hashtable.lib(hashtable.obj) : error LNK2019: unresolved external symbol _pool_new referenced in function _hashTableGrowSpaceOpt
00:15:47.876  ../../../bcvrelationships.dll : fatal error LNK1120: 11 unresolved externals
00:15:47.876  make[3]: *** [../../../makelib/targets.mk:464: ../../../bcvrelationships.dll] Error 96
00:15:47.876  make[2]: *** [makefile:479: bcvrelationships] Error 2
```

**Comment by keithc-ca** [comment]
For the record, builds and sanity testing were successful for jdk 8, 11 & 13 on pLinux.

**Comment by sharon-wang** [comment]
@keithc-ca testing some fixes for win32 locally - I'll let you know once I push the changes

**Comment by sharon-wang** [comment]
@keithc-ca Added `j9avl` and `j9pool` to link libraries. Personal testing for win32 passed.

**Comment by keithc-ca** [comment]
Jenkins test sanity win32 jdk8

File: test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/TestClassRelationshipVerifier.java [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+			if (!isRelationshipRecorded("B", "C", classLoader)) {
+				Assert.fail("relationship was not recorded");
+			}
+		} catch (InvocationTargetException e) {
+			throw e.getCause();
+		}
+	}
+
+	/**
+	 * Neither source, not target classes are loaded.
```
Comment (sharon-wang) [inline test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/TestClassRelationshipVerifier.java]: typo: not --> nor

Comment (keithc-ca) [reply test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/TestClassRelationshipVerifier.java]: Reviewing your own PRs, I see! That's great, but please let the win32 build finish before pushing anything new.

**Comment by sharon-wang** [comment]
@keithc-ca tests passed on windows - will be adding the last fix

**Comment by pshipton** [comment]
Causing https://github.com/eclipse/openj9/issues/7362
Reverting until fixed.
